lib/src/main/java/org/exoplatform/social/addons/updater/IdentityMigrationService.java

```diff
@@ -276,19 +277,25 @@ public void doRemove() throws Exception {
     } finally {
       MigrationContext.setIdentitiesCleanupFailed(identitiesCleanupFailed);
       if (identitiesCleanupFailed.size() > 0) {
-        LOG.warn("Cleanup failed for " + identitiesCleanupFailed.size() + " identities");
+        LOG.warn("Cleanup failed for {} identities", identitiesCleanupFailed.size());
       }
-      LOG.info(String.format("| / END::cleanup Identities migration for (%s) identity consumed %s(ms)", offset, System.currentTimeMillis() - t));
+      LOG.info("| / END::cleanup ({}) identities duration_ms={}", offset, System.currentTimeMillis() - t);
     }
   }
 
   private Identity migrateIdentity(Node node, String jcrId) throws Exception {
     String providerId = node.getProperty("soc:providerId").getString();
     String remoteId = node.getProperty("soc:remoteId").getString();
+    String name = node.getName();
+
+    // Ensure the nodes are coherents
+    if (!name.equals("soc:" + remoteId)) {
+      throw new Exception(String.format("Identity has different name and remote_id provider_id=%s name=%s remote_id=%s", providerId, remoteId, name));
```

Comment:
For me, the option 2 is better but harder to implement. If one identity was failed in migration, his activities/connections will not be migrated and beside that, some other's activities/connections which refer to him will not be migrated also. 
I still feel some risk on this way, we have to test carefully, but we could try to implement if you prefer

Anyway, I think we should to find the way to fix data to continue migrate that identity. Does it need the consistence between node name and remoteId? the remoteId isn't enough?